apps/web/src/scenes/NftDetailPage/NftDetailPageFallback.tsx

```diff
@@ -1,14 +1,47 @@
 import Skeleton from 'react-loading-skeleton';
 import styled from 'styled-components';
 
-import breakpoints from '~/components/core/breakpoints';
+import breakpoints, { size } from '~/components/core/breakpoints';
 import { HStack, VStack } from '~/components/core/Spacer/Stack';
+import { useNftPreviewFallbackState } from '~/contexts/nftPreviewFallback/NftPreviewFallbackContext';
+import { useIsMobileOrMobileLargeWindowWidth } from '~/hooks/useWindowSize';
+import {
+  DESKTOP_TOKEN_DETAIL_VIEW_SIZE,
+  MOBILE_TOKEN_DETAIL_VIEW_SIZE,
+} from '~/shared/utils/fitDimensionsToContainer';
+
+type Props = {
+  tokenId: string;
+};
+
+export default function NftDetailPageFallback({ tokenId }: Props) {
+  const { cachedUrls } = useNftPreviewFallbackState();
+
+  const hasPreviewUrl = cachedUrls[tokenId]?.type === 'preview';
+  const dimensions = cachedUrls[tokenId]?.dimensions;
+  const isMobileOrMobileLarge = useIsMobileOrMobileLargeWindowWidth();
+
+  const TOKEN_SIZE = isMobileOrMobileLarge
+    ? MOBILE_TOKEN_DETAIL_VIEW_SIZE
+    : DESKTOP_TOKEN_DETAIL_VIEW_SIZE;
 
-export default function NftDetailPageFallback() {
   return (
     <StyledFullPageLoader>
-      <StyledNftSkeleton />
-      <VStack gap={24}>
+      {hasPreviewUrl ? (
+        <VisibilityContainer>
+          <StyledImageWrapper className={'visible'}>
```

Comment:
we usually use props to pass in a styled state, ie `isVisible={isVisible}` , is there a special need to use a classname here?